src/smartcontracts/code.ts

```diff
@@ -15,6 +20,13 @@ export class Code {
         return new Code(code.toString("hex"));
     }
 
+    /**
+     * Creates a Code object from a hex-encoded string.
+     */
+    static fromHex(hex: string): Code {
```

Comment:
is this really needed? Since it basically is the same as the constructor?

Reply:
The constructor is currently private, and it could be made public indeed

Personally, I would still go with this implementation because it hides the internal representation of values, is more readable and provides more control over how code objects are created (e.g. validation added later)